### PR TITLE
🎄 2017 Winter Holidays update

### LINKED ISSUE #338: Automatic scroll-to-bottom while recording Events
When recording Events, the latest event quickly moves out of sight because the are added to the bottom of the list, but the scroll position is not updated accordingly.

image

It would be great to have an option to either automatically scroll to the bottom of the list as events come in or reverse the event list so that newer events are added to the top instead.

Cheers

### PR DESCRIPTION
**Pull request by Akryum**
This PR contains merges, fixes and minor changes for the following PRs:

- #314
- #413
- #418
- #454
- #470
- #471
- #472
- #474
- #477
- #478
- #479
- #481
- #483
- #484
- #488

Fix #338

### CODE REVIEW

**Comment by LinusBorg** [comment]
Holy Maccaroni!! This is incredible! 😍

**Comment by crswll** [comment]
Just a heads up. I needed to change this line https://github.com/Akryum/vue-devtools/blob/master/shells/chrome/webpack.config.js#L6 to this to get it to build properly.

```js
target: { chrome: 52, firefox: 48 },
```

File: src/devtools/views/components/ComponentInspector.vue [hunk 1]
```diff
@@ -49,7 +49,7 @@ export default {
   data () {
     return {
       filter: '',
-      isChrome: typeof chrome !== 'undefined' && chrome.devtools
+      isChrome: typeof chrome !== 'undefined' && !!chrome.devtools
```
Comment (Akryum) [inline src/devtools/views/components/ComponentInspector.vue]: Quick Edit boolean! :smile_cat:

File: docs/open-in-editor.md [hunk 2]
```diff
@@ -6,30 +6,34 @@ Nuxt supports this feature out-of-the-box. Make sure to be in debug mode.
...
-    app.use('/__open-in-editor', openInEditor({
-      editor: 'code'
-    }))
+    app.use('/__open-in-editor', openInEditor())
   }
 }
 ```
 
-3. You can change `'code'` with the editor you are using, see the [supported editors list](https://github.com/lahmatiy/open-in-editor#editor).
+3. The editor to launch is guessed. You can also specify the editor app with the `editor` option. See the [supported editors list](https://github.com/lahmatiy/open-in-editor#editor).
```
Comment (anteriovieira) [inline docs/open-in-editor.md]: Supported editors was  added in the launch-editor package.

https://github.com/yyx990803/launch-editor#supported-editors

**Comment by yyx990803** [comment]
Absolutely amazing job @Akryum, every feature works flawlessly!!!

**Comment by fixmycode** [comment]
Chrome is notifying me that this release requires new permissions, there should be a note somewhere explaining why.

**Comment by dttvinh** [comment]
Seems like the new permission has been removed in the latest commit. Looking forward to a new version soon :+1: .

**Pull request merged**